Display Enrolled Courses instead of Course Progress

merkushin opened the pull request:
Fixes part of #4958 

### Changes proposed in this Pull Request

* Display Enrolled Courses instead of Course Progress

### Testing instructions

* Create a few courses, add a few students and enroll courses.
* Check if you can see all the enrolments on the Students page.
* Use bulk actions to remove/add some courses for students.
* Check if you see those changes on the Students page.

### Screenshot / Video

https://user-images.githubusercontent.com/329356/164323296-0ac2e63f-9805-4e48-b4d4-fea64ed4e4ad.mp4

merkushin committed:
Display enrolled courses instead of course progress

merkushin committed:
Add tests for single row with enrolled courses

merkushin committed:
Fix tests with enrolments

merkushin committed:
Add short description for the class property

merkushin committed:
Declare $html_items variable before usage

LavaGolem commented:
Works as expected: 
1. Removed from course selected - course removed from Enrolled for a student
2. Add to course selected - course appears in Enrolled for a student

Imran92 commented on includes/admin/class-sensei-learners-admin-bulk-actions-view.php:
```diff
@@ -170,7 +179,7 @@ protected function get_row_data( $item ) {
 			);
 		} else {
 			$learner            = $item;
-			$courses            = $this->get_learner_courses_html( $item->course_statuses );
```
As we are not using the course_statuses field anymore, can we stop fetching it from the database in the Sensei_Db_Query_Learners::build_query function or make it optional? As a group concat operation is performed to prepare the data for this column. WDYT? Also if you decide to work on it, can you also take a look at the left join that we do in the build_query method with wp_comment table to see if we need it anymore for this page?

merkushin replied:
The problem is that it is public, so I am not sure if it is okay to remove part of data.
The other reason for me to keep it "as is" was that we will consider displaying statuses of courses (completed or not) soon.

merkushin replied:
I'll double-check the usage of the query is sensei, and, as an alternative, to keep code compatible replace `course_statuses` with empty string. Same for `course_count` if it isn't used anywhere.

merkushin replied:
@Imran92 Fixed here: https://github.com/Automattic/sensei/pull/5043/commits/4d72a99320f07ccc131446229db2012ee0cb010b

Imran92 commented on tests/unit-tests/admin/test-class-sensei-learners-admin-bulk-actions-view.php:
```diff
@@ -0,0 +1,159 @@
+	}
+
+	/**
+	 * Tests that the user gets redirected when the action is invalid.
```
Not a part of your changes I think :p But feel free to fix the comment

merkushin replied:
Fixed here: https://github.com/Automattic/sensei/pull/5043/commits/76f0b70eb91d2aadcb2d930e23471242f16591ca

merkushin committed:
Update comment for the test

merkushin committed:
Remove join of comments table to collect course progress

We don't use course progress anymore as we switched to course enrolments. We still return empty string for course_statuses field and zero for course_count for backward compatibility.

merkushin committed:
Fix linter issue (formatting)

merkushin committed:
Merge branch 'feature/student-management' into add/enrolled-courses

Imran92 reviewed (approved):
Looks good to me 👍
